### ISSUE TITLE
MathML to Docx

### ISSUE THREAD

**Issue by KaioSpa** [PHPOffice/PHPWord#2559]
Hey, I'm lost here and really need help. I'm sorry if it's obvious and I'm just blind...

I'm trying to convert MathML formulas to docx. I have seen that a merge was made, but I was not able to implement it.

When I try to convert directly to ODText, the formula does not appear, and it just shows the numbers without the formula (10^20 appears as 1020).
```
$phpWord = new PhpWord();
    $section = $phpWord->addSection();
    Html::addHtml($section, $html, false, false);

    // Save the resulting document to a temporary file
    $tempDoc = tempnam(sys_get_temp_dir(), 'PHPWord');
    $phpWord->save($tempDoc, 'Word2007');
```

And when trying to use the Formula class, it just returns empty:

`$math = new Formula(new Math($mathML));`

An example of my MathML element (I extracted the element to try to use the Formula class):
`$mathML=<math xmlns="http://www.w3.org/1998/Math/MathML"><mn>4</mn><mi>x</mi><mo>-</mo><mfrac><mrow><mn>3</mn><mo>-</mo><mi>x</mi></mrow><mn>2</mn></mfrac><mo>=</mo><mn>21</mn><mo>&#160;</mo><mo>&#160;</mo><mo>&#160;</mo><mi>e</mi><mo>&#160;</mo><mo>&#160;</mo><mo>&#160;</mo><mfrac><mrow><mi>y</mi><mo>-</mo><mn>3</mn></mrow><mn>2</mn></mfrac><mo>=</mo><mfrac><mn>1</mn><mrow><mi>y</mi><mo>+</mo><mn>2</mn></mrow></mfrac><mo>+</mo><mfrac><mn>3</mn><mn>2</mn></mfrac></math>`

Can someone help?

**Comment by Opussci** [comment]
Hi @KaioSpa ,
have you managed to solve this issue?

I am also very much interested in writing MathML to docx files, can anybody help with this?

**Comment by Progi1984** [comment]
@KaioSpa & @Opussci 

It's fixed in develop branch with #2645.

Code : 
```


// Read contents
$name = basename(__FILE__, '.php');

$source = __DIR__ . "/resources/{$name}.docx";

$mathML='<math xmlns="http://www.w3.org/1998/Math/MathML">
<mn>4</mn><mi>x</mi><mo>-</mo>
<mfrac><mrow><mn>3</mn><mo>-</mo><mi>x</mi></mrow><mn>2</mn></mfrac><mo>=</mo><mn>21</mn><mo>&#160;</mo><mo>&#160;</mo><mo>&#160;</mo><mi>e</mi><mo>&#160;</mo><mo>&#160;</mo><mo>&#160;</mo><mfrac><mrow><mi>y</mi><mo>-</mo><mn>3</mn></mrow><mn>2</mn></mfrac><mo>=</mo><mfrac><mn>1</mn><mrow><mi>y</mi><mo>+</mo><mn>2</mn></mrow></mfrac><mo>+</mo><mfrac><mn>3</mn><mn>2</mn></mfrac></math>';

$reader = new \PhpOffice\Math\Reader\MathML();
$math = $reader->read($mathML);

$phpWord = new \PhpOffice\PhpWord\PhpWord();
$section = $phpWord->addSection();
$section->addFormula($math);

$phpWord->save($source, 'Word2007');
```

![image](https://github.com/user-attachments/assets/7461df30-51c3-47f9-89da-ce4917955dcd)

**Comment by Progi1984** [comment]
@Opussci & @KaioSpa This issue has been fixed by a maintainer. You can help him by sponsoring him through [Github sponsors](https://github.com/sponsors/Progi1984).
